### ISSUE TITLE
SC2154 false positive in single/double quoted vars/aliases and inconsistencies

### ISSUE THREAD

**Issue by pstueck** [koalaman/shellcheck#2894]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or "online"): `0.9.0`, "online", VSCode Extension 0.35.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
alias newalias='_ugs=5 _ug=0 ; echo $_ugs $_ug'

```

#### Here's what shellcheck currently says:
```
SC2154 (warning): _ug is referenced but not assigned.
SC2154 (warning): _ugs is referenced but not assigned.
```

#### Here's what I wanted or expected to see:
no warning

----

#### On the other hand …
```sh

#!/bin/bash
_ugs=9
alias newalias='_ugs=5 ; echo $_ugs $((_ug=_ug+1))'

```
… is accepted, although I think, `_ug` should have caused an SC2154

**Comment by skull-squadron** [comment]
Similar false positives with _lower-cased_ variables in a trap statement despite being referenced. It doesn't happen with _upper-cased_ variable names.

```
In files/base/installer/texlive line 9:
trap 'e=$?; trap - EXIT; rm -rf "$TEXLIVE_TMP_DIR"; exit $e' EXIT
     ^-- SC2154 (warning): e is referenced but not assigned.
```
